cs/src/Connections/IRelayClient.cs

```diff
@@ -27,8 +27,32 @@ internal interface IRelayClient
     /// </summary>
     TraceSource Trace { get; }
 
+    /// <summary>
+    /// Get the tunnel connection role, either "client", or "host".
+    /// </summary>
+    string ConnectionRole { get; }
+
+    /// <summary>
+    /// Get relay client dispose token.
+    /// </summary>
+    CancellationToken DisposeToken { get; }
+
+    /// <summary>
+    /// Start connecting relay client.
+    /// </summary>
+    /// <exception cref="ObjectDisposedException">If relay client is disposed.</exception>
+    void StartConnecting();
```

Comment:
Does this do anything other than change the connection status?

Reply:
Yep. This change is needed to isolate where the connection status is set to `Connecting`.